python/cudf/cudf/core/index.py

```diff
@@ -1411,24 +1466,196 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class BaseNumericIndex(GenericIndex):
+    """Immutable, ordered and sliceable sequence of labels.
+    The basic object storing row labels for all cuDF objects.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype,
+            but not used.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+    """
+
+    def __new__(cls, data=None, dtype=None, copy=False, name=None):
+
+        out = Frame().__new__(cls)
+        dtype = index_to_dtype[cls]
+        if copy:
+            data = column.as_column(data, dtype=dtype).copy()
+
+        kwargs = _setdefault_name(data, name=name)
+
+        data = column.as_column(data, dtype=dtype)
+
+        out._initialize(data, **kwargs)
+
+        return out
+
+
+class Int8Index(BaseNumericIndex):
+    __doc__ = BaseNumericIndex.__doc__
+
+    def __init__(cls, data=None, dtype=None, copy=False, name=None):
+        pass
+
+
+class Int16Index(BaseNumericIndex):
+    __doc__ = BaseNumericIndex.__doc__
+
+    def __init__(cls, data=None, dtype=None, copy=False, name=None):
```

Comment:
Does it make sense to have a `dtype` arg here? What happens if one passes `dtype="float64"` to the constructor of an `Int16Index`?

Reply:
I have already mentioned in the doc that dtype is not being used and it is just present to be in sync with pandas.

Reply:
We actually need the dtype arg here: say someone does something like `Int64Index([1, 2, 257], dtype='int8')` the expected behavior is to overflow to `[1, 2, 1]`.

Reply:
I was thinking same, but not sure if it has any valid use case.

Reply:
And don't think even pandas is doing anything, 
```
>>> import pandas as pd
>>> pd.Int64Index([1, 2, 257], dtype='int8')
Int64Index([1, 2, 257], dtype='int64')
```

Reply:
What version of Pandas is that testing with? As of 1.0 it looks to be behaving as expected now.

```
In [42]: pd.Int64Index([1, 2, 257], dtype='int8')
Out[42]: Int64Index([1, 2, 1], dtype='int64')
```

Reply:
```
>>> import pandas as pd
>>> pd.__version__
'0.25.3'
```